src/main/webapp/file.xhtml

```diff
@@ -337,7 +337,7 @@
                                 </ui:include>
                             </p:tab>
                             <p:tab id="previewTab" title="#{bundle['file.previewTab.header']}" 
-                                   rendered="#{FilePage.toolsWithPreviews.size() > 0 and FilePage.previewAllowed}">
+                                   rendered="#{FilePage.toolsWithPreviews.size() > 0 and fileDownloadHelper.isPreviewAllowed(FilePage.fileMetadata)}">
```

Comment:
FWIW: At QDR, I had changed this to fileDownloadHelper.canDownloadFile() - both here and line 402 below. A couple things:
- Not sure if that's better, but I think it works with PrivateUrls, so I'm not sure the isPreviewAllowed method is needed any more? (It's possible that there is some other difference that makes this work at QDR).
- Independently, I think you'd want the same change for 402?
- Since the canDownloadFile method checks for the embargo state (which is called by isPreviewAllowed too), I think the logic should be OK w.r.t. embargoes (embargoes only stop people without permissions (and don't let you request access, etc. - they don't override if you can see the draft dataset, download the files etc as a curator, privateURL user, etc.)

Reply:
Thanks for the comments Jim. I'm not sure if I am the one to reply but will anyway.

-  the fileDownloadHelper.isPreviewAllowed() prevents users who don't have permission to view/download a file from previewing it but allowing those who do have permission to preview it. So it's needed imo but maybe I am missing something for you to say it may not be necessary?
- the fileDownloadHelper.isPreviewAllowed() is already called at line 402 (if that's what you meant).
- I tested the changes with embargoed files and it 'works' but I would just like the Dataverse team to verify the behaviour is as expected.
- testing it with privateUrl would also be beneficial.

Reply:
I think what @qqmyers is suggesting is that we could get rid of fileDownloadHelper.isPreviewAllowed() all together and just use fileDownloadHelper.canDownloadFile() in its place. (i.e. a user can preview if and only if they can download).

Can we confirm that is definitely the case? It sounds so to me? If so it may still be worth modifying fileDownloadHelper.canDownloadFile with the user check from isPreviewAllowed, since that seems less expensive than checking permissions.

Reply:
I agree. I missed that 402 was already doing this - just saw that it wasn't changed in the PR - so a change there is only needed if we change to canDownloadFile. I see @scolapasta's point that the isPreview method is more efficient for PrivateURLs since it bypasses the checks. It seems like that same logic could/would make sense for canDownloadFile as well - PrivateURL users can always download files so we could do the check for them in canDownloadFile and make all of those checks more efficient for privateURL users too (and if that check works in the isPreview method now, it woudl work in canDownloadFile even if I'm wrong about the permission checks. 

Again, I think this is all minor but having additional methods around that end up doing the same thing are just a pain when making other changes so it would be nice to get rid of it with a small change here.

Reply:
Thanks for the comments. And I agree that these issues are minor in the grand scheme of things and other code that needs to be implemented but the Preview functionality should work. Reducing the number of calls that essentially do the same thing, or result in the same thing, is also beneficial. 

I can add the privateUrl user check to canDownloadFile() and replace the calls to isPreviewAllowed() and delete the latter altogether. And then test again.

Reply:
So I added the user is PrivateUrlUser check to canDownloadFile() and then changed all previous calls to isPreviewAllowed() to canDownloadFile() and deleted isPreviewAllowed() from FileDownloadHelper.

I believe the previewTable is now rendered as it should but will leave it to @sekmiller to verify the behaviour is as expected for privateUrls, embargoes, restricted/unrestricted, etc.